src/aiida_quantumespresso/workflows/xspectra/core.py

```diff
@@ -0,0 +1,910 @@
+# -*- coding: utf-8 -*-
+"""Workchain to compute the X-ray absorption spectrum for a given structure.
+
+Uses QuantumESPRESSO pw.x and xspectra.x, requires ``AiiDA-Shell`` to run ``upf2plotcore.sh``.
+"""
+import pathlib
+from typing import Optional, Union
+
+from aiida import orm
+from aiida.common import AttributeDict
+from aiida.engine import ToContext, WorkChain, calcfunction, if_
+from aiida.orm.nodes.data.base import to_aiida_type
+from aiida.plugins import CalculationFactory, DataFactory, WorkflowFactory
+import yaml
+
+from aiida_quantumespresso.utils.mapping import prepare_process_inputs
+from aiida_quantumespresso.workflows.protocols.utils import ProtocolMixin, recursive_merge
+
+PwCalculation = CalculationFactory('quantumespresso.pw')
+PwBaseWorkChain = WorkflowFactory('quantumespresso.pw.base')
+XspectraBaseWorkChain = WorkflowFactory('quantumespresso.xspectra.base')
+XyData = DataFactory('array.xy')
+
+
+@calcfunction
+def get_all_spectra(**kwargs):
```

Comment:
Could you please move each of these calcfunctions to a separate file in the directory `aiida_quantumespresso/calculations/functions/xspectra`. You need to create the directory first.

Reply:
Also would a slightly more specific name be useful? Maybe something like `merge_spectra`?